docs/src/modules/ROOT/pages/using-timefold-solver/modeling-planning-problems.adoc

```diff
@@ -1367,6 +1367,121 @@ public class Customer {
     public void setNextCustomer(Customer nextCustomer) {...}
 ----
 
+[#tailChainVariable]
+=== Updating tail chains
+
+The annotation `@CascadingUpdateListener` provides a built-in listener that updates a set of connected elements.
+Timefold Solver triggers a user-defined logic whenever the inverse relation or previous element changes.
+Moreover, it automatically propagates changes to the next elements when the value of the related shadow variable changes.
+
+The planning entity side has a genuine list variable:
+
+[source,java]
+----
+@PlanningEntity
+public class Vehicle {
+
+    @PlanningListVariable
+    public List<Customer> getCustomers() {
+        return customers;
+    }
+
+    public void setCustomers(List<Customer> customers) {...}
+}
+----
+
+On the element side:
+
+[source,java]
+----
+@PlanningEntity
+public class Customer {
+
+    @InverseRelationShadowVariable(sourceVariableName = "customers")
+    private Vehicle vehicle;
+    @PreviousElementShadowVariable(sourceVariableName = "customers")
+    private Customer previousCustomer;
+    @NextElementShadowVariable(sourceVariableName = "customers")
+    private Customer nextCustomer;
+    @CascadingUpdateListener(targetMethodName = "updateArrivalTime", sourceVariableName = "vehicle")
+    @CascadingUpdateListener(targetMethodName = "updateArrivalTime", sourceVariableName = "previousCustomer")
+    private LocalDateTime arrivalTime;
+
```

Comment:
The very first example shouldn't be this complicated.
Reduce it to the bare minimum code. You can explain the extra features later.